switch to vite and vitest

ToppleTheNun opened the pull request:
### Description

<!-- A brief description of the changes made with this pull request.-->

vite is much faster than create-react-app
and create-react-app is also stagnant.

a lot of the changes are a result of an auto-import cleanup that
was done to help with the vitest migration.

### Testing

<!-- How can the reviewer test your changes (if applicable)? -->

You need to pull it and test it yourself. :)

Woliance committed:
WIP Guide

Woliance committed:
WIP Guide

Woliance committed:
Making a savestate

Not finished

Woliance committed:
Making a savestate

Not finished

Woliance committed:
Pushing pending changes to fork

Woliance committed:
Pushing pending changes to fork

ToppleTheNun committed:
switch to pnpm

Woliance committed:
Removing <Trans> and renaming CastLinkNormalizer

<Trans rights>

ToppleTheNun committed:
switch to vite and vitest

vite is much faster than create-react-app
and create-react-app is also stagnant.

emallson commented on Dockerfile.dev:
```diff
@@ -6,6 +6,6 @@ RUN npm i -g pnpm
 
 COPY . /usr/src/wowanalyzer/
 
-EXPOSE 3000
```
would prob be good to keep the development port the same if possible to easily/reliably configure. we have enough stuff changing, don't need to add more.

ToppleTheNun replied:
I will repoint everything to 3000

emallson commented on vite.config.ts:
```diff
@@ -0,0 +1,102 @@
+      name: 'vite-plugin-wowanalyzer-index-html-inject-ga',
+      transformIndexHtml: (html) =>
+        process.env.VITE_ENABLE_GA === 'true'
+          ? html.replace('</head>', GOOGLE_ANALYTICS_SCRIPT)
```
a little spicy, but clever

emallson reviewed (commented):
i have only skimmed this, excited to try it out

ToppleTheNun committed:
fix lingui imports

ToppleTheNun committed:
switch to swc over babel (even faster!)

also updates to lingui 4.x to use the
swc plugin.

ToppleTheNun committed:
add changelog entry

ToppleTheNun committed:
finish up lingui update

ToppleTheNun committed:
remove make-plural

ToppleTheNun committed:
fix PremiumPage export

ToppleTheNun committed:
add lazyRetry to see if it helps with build

Pewtro committed:
[Hunter] Add deathblow buff to the timeline

Pewtro committed:
Merge pull request #6438 from Pewtro/hunter/db-buff

[Hunter] Add deathblow buff to the timeline

ToppleTheNun committed:
make Soul Fragment not a cast

ToppleTheNun committed:
Merge pull request #6439 from ToppleTheNun/vdh-soul-fragments

make Soul Fragment not a cast

Woliance committed:
Adding CDR effects

RP, RD, Valkyr

Woliance committed:
Merge branch 'ppala' of https://github.com/Woliance/WoWAnalyzer into ppala

Woliance committed:
Update BuilderUse.tsx

Woliance committed:
Delete Unused File

Woliance committed:
Adding sentinel support

Woliance committed:
Minor Inmost Light change

Woliance committed:
Update CHANGELOG.tsx

avilene committed:
create internetexplorer slice

Pewtro committed:
[Dwarf] Fix dwarf racial with pets

Pewtro committed:
It also works for non-permanent pets

Pewtro committed:
Merge pull request #6441 from Pewtro/bugfix/dwarf-racial

[Dwarf] Fix dwarf racial with pets

Woliance committed:
Cooldown Efficiency Update

Pewtro committed:
[Core] Add an optional flag that will make the crit effect be counted double when calculating value

This is at best a temporary solution for https://github.com/WoWAnalyzer/WoWAnalyzer/issues/6442 because the hook is still not accurately reflecting the difference between the different crit effects, but it doesn't seem to matter for Might of the Mountain.

avilene committed:
migrate charactersById

ToppleTheNun committed:
add background images for Amirdrassil bosses

ToppleTheNun committed:
Merge pull request #6444 from ToppleTheNun/amirdrassil-backgrounds

add background images for Amirdrassil bosses

avilene committed:
convert combatants

smvoss committed:
windwalker: fix FoF channeling and APL

FoF was breaking the APL due to both the cast and beginchannel being
checked, causing the second one to raise a failure. Add property to
track the previous cast in the APL as well as previous beginchannel so
that we can watch for duplicates that happen in either direction.

Signed-off-by: Sam Voss <sam.voss@gmail.com>

avilene committed:
slice language

avilene committed:
slice modal

avilene committed:
slice report

avilene committed:
force typing on nullable initialstates

avilene committed:
slice ignorepatchcode

avilene committed:
slice reportHistory

avilene committed:
slice specsIgnored

avilene committed:
slice tooltips

avilene committed:
slice user

avilene committed:
fix logout import

avilene committed:
remove all actions (they are now unused!)

avilene committed:
add changelog entry

avilene committed:
Merge branch 'dragonflight' into feat/rtk-slices

avilene committed:
fix typo in import

emallson committed:
Merge branch 'dragonflight' into bugfix/crit-effect-bonus-maybe

emallson committed:
Merge pull request #6445 from smvoss/savoss/feature/windwalker-fof-channeling-apl-fix

windwalker: fix FoF channeling and APL

emallson committed:
fix up some `state = X` that don't do what we want

emallson committed:
remove console.log

emallson committed:
Merge pull request #6443 from Pewtro/bugfix/crit-effect-bonus-maybe

[Core] Add an optional flag to CritEffectBonus for effects that affect total damage

avilene committed:
Merge branch 'dragonflight' into feat/rtk-slices

ToppleTheNun committed:
fix @lingui/cli version

Woliance committed:
Update Active Mitigation Description

emallson committed:
small updates to vite/vitest config

- ignore .direnv (this is a cache folder for direnv, mine happens to
  have....a lot of files and runs the system out of file handles when
  vite tries to watch it lmao)
- disable canvas in tests, suppress jsdom warning about it, re-enable
  threads for tests
- add jsdom as a dev-dependency (it was pulled in by jest-canvas-mock i
  think?)

ToppleTheNun committed:
Merge pull request #6440 from Woliance/ppala

Prot Paladin Guide Update

smvoss committed:
windwalker: add SOTW to core abilities

Also add graphs to show Fists of Fury hit-ticks distribution under
statistics frame.

Signed-off-by: Sam Voss <sam.voss@gmail.com>

ToppleTheNun committed:
Merge pull request #6446 from avilene/feat/rtk-slices

Move reducers and actions into slices

ToppleTheNun committed:
update vitest

ToppleTheNun committed:
enable toggling between SWC and Babel

smvoss committed:
windwalker: add ability which clipped FoF to details

Signed-off-by: Sam Voss <sam.voss@gmail.com>

emallson committed:
Merge pull request #6448 from smvoss/savoss/feature/windwalker-fof-stats-update

windwalker: add SOTW to core abilities, and more FoF info

emallson committed:
this shouldn't have been happening. oops.

[skip ci]

emallson committed:
Merge pull request #6450 from emallson/ga-dont-dockerize-merge-queue

stop publishing merge queue as tagged docker images
